### PR TITLE
Ros2 plugin custom naming

### PR DESCRIPTION
**Pull request by owny990312**

### CODE REVIEW

File: src/plugins/ros2_plugin/ros2_rpc_backend.cc [hunk 1]
```diff
@@ -243,6 +252,56 @@ rclcpp::QoS Ros2RpcBackend::GetQos(const Options::QosOptions& qos_option) {
...
+                                 placeholder_match.length(),
+                                 match[index].str());
+      search_start = replaced_func_name.cbegin() + placeholder_match.position() + match[index].length();
+    } else {
+      AIMRT_WARN("Regex placeholder index out of range, index: {}, match size: {}", index, match.size());
+      return input_string;
+    }
+  }
+
+  // in AimRT, func_name must be start with "<msg_typr>:/", if not, add it.
```
Comment (AimRT) [inline src/plugins/ros2_plugin/ros2_rpc_backend.cc:294]: msg_type

File: src/plugins/ros2_plugin/ros2_rpc_backend.cc [hunk 2]
```diff
@@ -243,6 +252,56 @@ rclcpp::QoS Ros2RpcBackend::GetQos(const Options::QosOptions& qos_option) {
...
+  if (matching_rule.empty()) {
+    AIMRT_WARN("You have not set matching rule for remapping, please add 'func_name' option in yaml file.");
+    return input_string;
+  }
+
+  std::regex re(matching_rule);
+  std::smatch match;
+
+  // in this case, means there are no matched, return the original inoput string
+  if (!std::regex_search(input_string, match, re)) {
```
Comment (AimRT) [inline src/plugins/ros2_plugin/ros2_rpc_backend.cc:271]: use try catch

File: src/plugins/ros2_plugin/ros2_rpc_backend.cc [hunk 3]
```diff
@@ -243,6 +252,56 @@ rclcpp::QoS Ros2RpcBackend::GetQos(const Options::QosOptions& qos_option) {
   return qos;
 }
 
+std::string Ros2RpcBackend::GetRemappedFuncName(const std::string& input_string, const std::string& matching_rule, const std::string& remapping_rule) {
```
Comment (AimRT) [inline src/plugins/ros2_plugin/ros2_rpc_backend.cc:255]: add UT for this method

File: src/plugins/ros2_plugin/ros2_rpc_backend.h [hunk 4]
```diff
@@ -115,14 +120,27 @@ class Ros2RpcBackend : public runtime::core::rpc::RpcBackendBase {
...
+
  private:
   static bool CheckRosFunc(std::string_view func_name) {
     return (func_name.substr(0, 5) == "ros2:");
   }
 
   std::string GetRealRosFuncName(std::string_view func_name) {
-    return rclcpp::extend_name_with_sub_namespace(
-        std::string(func_name.substr(5)), ros2_node_ptr_->get_sub_namespace());
+    std::vector<std::string_view> splited_vec = common::util::SplitToVec<std::string_view>(func_name, ":");
```
Comment (wtudio) [inline src/plugins/ros2_plugin/ros2_rpc_backend.h:131]: simplify the code

**Pull request merged**